### PR TITLE
refactor(billing): use tanstack table instead of old one

### PR DESCRIPTION
**Pull request by yusualhashash**
## How has this been tested?

- [ ] Prod
- [x] Development

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] Added all translations, no hard coded text. 
- [ ] Updated the UI (if needed).
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by ertugrulcan-ays — DISMISSED** [review]

File: apps/web/src/app/[lang]/app/[type]/finance/billing/table-data.tsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        icon: Trash,
+        onConfirm: (row) => {
+          void deleteBillingApi(row.id || "").then((response) => {
+            if (response.type === "success") {
+              toast.success(response.message || languageData["Delete.Success"]);
+              router.refresh();
+            } else {
+              toast.error(response.message || languageData["Delete.Fail"]);
+            }
+          });
```
Comment (eren-ayasofyazilim) [inline apps/web/src/app/[lang]/app/[type]/finance/billing/table-data.tsx:151]: create a handleDelete function, so we can use it anywhere without additional code lines

File: apps/web/src/app/[lang]/app/[type]/finance/billing/table-data.tsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+              value: "PAID",
+            },
+            {
+              label: languageData["Form.status.SENT"],
+              value: "SENT",
+            },
+            {
+              label: languageData["Form.status.UNFINISHED"],
+              value: "UNFINISHED",
+            },
```
Comment (eren-ayasofyazilim) [inline apps/web/src/app/[lang]/app/[type]/finance/billing/table-data.tsx:106]: You can make this shorter like this;
resim

Here is what you should do with your code:
  $PagedResultDto_BillingDto.properties.items.items.properties.status.enum.map(
    (x) => ({
      label: languageData["Form.status." + x],
      value: x,
  }))

File: apps/web/src/app/[lang]/app/[type]/finance/billing/[billingId]/form.tsx [hunk 3]
```diff
@@ -38,18 +39,15 @@ export default function Form({
...
       id: billingId,
       requestBody: data,
     });
-    if (response.type === "success") {
-      toast.success(languageData.finance["Billing.Update.Success"]);
-    } else {
-      toast.error(response.type + response.message || ["Billing.Update.Fail"]);
-    }
+    handlePutResponse(response, router);
   }
```
Comment (eren-ayasofyazilim) [inline apps/web/src/app/[lang]/app/[type]/finance/billing/[billingId]/form.tsx:51]: don't make it async, instread use then callback.

resim

File: apps/web/src/app/[lang]/app/[type]/finance/billing/table-data.tsx (hunk 1 above)
Comment (yusualhashash) [reply apps/web/src/app/[lang]/app/[type]/finance/billing/table-data.tsx]: done

File: apps/web/src/app/[lang]/app/[type]/finance/billing/[billingId]/form.tsx (hunk 3 above)
Comment (yusualhashash) [reply apps/web/src/app/[lang]/app/[type]/finance/billing/[billingId]/form.tsx]: done

**Review by eren-ayasofyazilim — DISMISSED** [review]
looks nice, thanks for changes

**Comment by eren-ayasofyazilim** [comment]
good luck on solving conflict  @yusualhashash

**Review by eren-ayasofyazilim — APPROVED** [review]

**Review by ertugrulcan-ays — APPROVED** [review]

**Pull request merged**